src/modules/networks/Networks.ts

```diff
@@ -25,152 +25,75 @@ function toCollection<T, K>(
-  undefined | { items: NetworkConfig[]; collection: Collection<NetworkConfig> }
->;
-
 export interface NetworkConfigMetaData {
   created: number;
   updated: number;
   origin: string;
 }
 
-function normalizeSources(
-  sources: EthereumChainSources | undefined
-): EthereumChainSourcesNormalized {
-  if (!sources) {
-    return {};
-  }
-  const ethereumChainSourcesNormalized: EthereumChainSourcesNormalized = {};
-  for (const [key, value] of Object.entries(sources)) {
-    const networkConfigs = value.ethereumChains.map((x) => x.value);
-    ethereumChainSourcesNormalized[key] = {
-      items: networkConfigs,
-      collection: toCollection(
-        networkConfigs,
-        (item) => item.chain,
-        (x) => x
-      ),
-    };
-  }
-  return ethereumChainSourcesNormalized;
-}
-
 export class Networks {
   private networks: NetworkConfig[];
-  ethereumChainSources?: EthereumChainSources;
-  private sourcesNormalized?: EthereumChainSourcesNormalized;
   private collection: { [key: string]: NetworkConfig | undefined };
   private collectionByEvmId: { [key: string]: NetworkConfig | undefined };
-  private originalCollection: { [key: string]: NetworkConfig | undefined };
-  private supportedByBackend: Set<string>;
+  private ethereumChainConfigs: EthereumChainConfig[];
   static purposeKeyMap = {
     sending: 'supports_sending',
     trading: 'supports_trading',
-    bridge: 'supports_bridge',
+    bridge: 'supports_bridging',
+    actions: 'supports_actions',
+    nfts: 'supports_nft_positions',
+    positions: 'supports_positions',
   } as const;
 
   constructor({
     networks,
-    ethereumChainSources,
+    ethereumChainConfigs,
   }: {
     networks: NetworkConfig[];
-    ethereumChainSources?: EthereumChainSources;
+    ethereumChainConfigs: EthereumChainConfig[];
   }) {
-    this.networks = networks
-      .filter((network) => Boolean(network.evm_id))
-      .sort((a, b) => localeCompareWithPriority(a.name, b.name, 'Ethereum'));
-    this.ethereumChainSources = ethereumChainSources;
-    const {
-      collection,
-      originalCollection,
-      collectionByEvmId,
-      sourcesNormalized,
-    } = this.prepare();
-    this.collection = collection;
-    this.originalCollection = originalCollection;
-    this.collectionByEvmId = collectionByEvmId;
-    this.sourcesNormalized = sourcesNormalized;
-    this.supportedByBackend = new Set(networks.map((n) => n.chain));
-  }
-
-  private prepare() {
-    const originalCollection = toCollection(
+    this.ethereumChainConfigs = ethereumChainConfigs;
+    this.networks = insertChainConfigs(networks, ethereumChainConfigs);
+    this.collection = toCollection(
       this.networks,
-      (network) => network.chain,
+      (network) => network.id,
       (x) => x
     );
-    const collectionByEvmId = toCollection(
+    this.collectionByEvmId = toCollection(
       this.networks,
-      (x) => x.external_id,
+      (x) => getChainId(x) || '',
```

Comment:
It's weird to have empty string as a key.
What do we do about such networks? I would suggest filtering them out.

In turn, `getChainId(x): string` should maybe even throw error when it cannot find a chainId?
And we can later introduce a `peekChainId(x): string | null` if we need to deliberately look for a network that has a chainId missing?